src/main/java/com/amplitude/Event.java

```diff
@@ -240,11 +240,18 @@ public JSONObject toJsonObject() {
       event.put(
           "user_properties",
           (userProperties == null) ? new JSONObject() : truncate(userProperties));
-      event.put("price", price);
-      event.put("quantity", quantity);
-      event.put("revenue", revenue);
-      event.put("productId", productId);
-      event.put("revenueType", revenueType);
+
+      boolean shouldLogRevenueProps = (revenue != null || (price != null && quantity > 0));
+      if (shouldLogRevenueProps) {
+        event.put("price", price);
+        event.put("quantity", quantity);
+        event.put(
+            "revenue",
+            (revenue != null && price != null && quantity > 0) ? quantity * price : revenue);
```

Comment:
I am a bit confused here, so if there exists revenue and price and quantity then we do our calculation?

Reply:
it’s how api v2 says.

https://developers.amplitude.com/docs/http-api-v2

revenue = price * quantity. If you send all 3 fields of price, quantity, and revenue, then (price * quantity) will be used as the revenue value. You can use negative values to indicate refunds.

Reply:
Oh, I just realized we send the payload to api v2 eventually. Will update this to if have revenue or have price and valid quantity will pass other revenue related values.

Reply:
`if have revenue or have price and valid quantity` this is already true when we get to this part right? What's our normal way to calculate this in SDK?